index.js

```diff
@@ -41,43 +43,55 @@ function checkGuess() {
 
   if (guess !== targetNumber) {
     if (guess < targetNumber) {
-      tooLowMessage.style.display = '';
+      if(guess <= 0){
+        belowZeroMessage.style.display = 'block';
+      }else {
+      tooLowMessage.style.display = 'block';
+    }
     } else {
-      tooLowMessage.style.display = '';
+      if(guess>=100){
+        aboveHundredMessage.style.display = 'block';
+      }else{
+      tooHighMessage.style.display = 'block';
     }
+    } 
 
     const remainingAttempts = maxNumberOfAttempts - attempts;
 
-    numberOfGuessesMessage.style.display = '';
+    numberOfGuessesMessage.style.display = 'block';
+    if(attempts===4){
+      numberOfGuessesMessage.innerHTML = `You guessed ${guess}. <br> ${remainingAttempts} guess remaining`;
+    } else{
     numberOfGuessesMessage.innerHTML = `You guessed ${guess}. <br> ${remainingAttempts} guesses remaining`;
-  }
+  }}
 
-  if (attempts ==== maxNumberOfAttempts) {
+  if (attempts === maxNumberOfAttempts) {
     submitButton.disabled = true;
     guessInput.disabled = true;
   }
 
   guessInput.value = '';
 
-  resetButton.style.display = '';
+  resetButton.style.display = 'block';
 }
 
 function hideAllMessages() {
-  for (let elementIndex = 0; elementIndex <= messages.length; elementIndex++) {
+  for (let elementIndex = 0; elementIndex < messages.length; elementIndex++) {
     messages[elementIndex].style.display = 'none';
   }
 }
 
-funtion setup() {
+function setup() {
   // Get random number
   targetNumber = getRandomNumber(1, 100);
   console.log(`target number: ${targetNumber}`);
 
   // Reset number of attempts
-  maxNumberOfAttempts = 0;
+  const maxNumberOfAttempts = 5;
```

Comment:
Fix: Unfortunately, this is still not doing what we expect. In this case, we don't want to reset the `maxNumberOfAttempts` (because it's always 5 regardless of a new or old game). But what else needs to be reset!? See screenshot below:
![Screenshot from 2023-07-10 15-44-52](https://github.com/butolinka/intro-to-programming-section-5/assets/22607722/87d40787-b802-4576-81e7-7ceb76fe36de)

Reply:
Matt, thank you! The answer was literally in front of me in the comment on line 89.

Reply:
Mariia, excellent work at finding the bug! Sometimes the solution is right there, but we often miss it. Don't worry though, it happens to **ALL** of us. For example, just recently, I was setting a value to be defined only for it become undefined 10 lines later in the code -- it took me way too long to figure out that I wrote some code to reset that value to be undefined)! :laughing: